tests/src/Simple/ReadyToRunTokens/Program.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Also, was something actually failing in this test before your fix, or it is just one of those tests that one needs to manually inspect in a debugger to see if things are broken?

Reply:
The bugs I hit while debugging your test were as follows:

1) We were generating specific, not canonical, version for Caller1<string>.Compare.
2) The emission of the instance entrypoint table was broken in many ways (in fact it was missing completely and those parts of the code that existed had bugs of their own like early unification of signature / fixup blobs that are generally not available before we have actually compiled all the methods).
3) Handling of __Canon in signatures was missing (naturally, as the canonicalization itself was absent).

Without these changes the test was technically passing however it wasn't using the CPAOT code so there was no way to verify that the emission of the generic instantiations was correct.

I fixed all the codegen issues long before I made the hashcodes work so that, once the hashcodes clicked in, the R2R code was immediately passing. [Perhaps not that surprising as, by that time, it was more or less 1:1 identical with Crossgen code.]